addons/website_project/security/project_security.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+        <!-- Portal -->
+        <record id="project.portal_project_rule" model="ir.rule">
+            <field name="name">Project: portal users: portal or following</field>
+            <field name="model_id" ref="project.model_project_project"/>
+            <field name="domain_force">[
+                '|',
+                    '&amp;',
+                        ('privacy_visibility', '=', 'portal'),
+                        ('message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id]),
+                    '&amp;',
+                        ('privacy_visibility', '=', 'followers'),
+                        ('message_partner_ids', 'in', [user.partner_id.id])
+            ]</field>
+            <field name="groups" eval="[(4, ref('base.group_portal'))]"/>
+        </record>
+
+        <record id="project.portal_task_rule" model="ir.rule">
+            <field name="name">Project/Task: portal users: (portal and colleagues following) or (followers and following)</field>
+            <field name="model_id" ref="project.model_project_task"/>
+            <field name="domain_force">[
+            '|',
+                '|',
+                    '&amp;',
+                        ('project_id.privacy_visibility', '=', 'portal'),
+                        ('project_id.message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id]),
+                    '&amp;',
+                        ('project_id.privacy_visibility', '=', 'followers'),
+                        ('project_id.message_partner_ids', 'in', [user.partner_id.id]),
+                '&amp;',
+                    # on employee project can receive messages but not access the object
+                    ('project_id.privacy_visibility', '!=', 'employees'),
+                    ('message_partner_ids', 'in', [user.partner_id.id])
+            ]</field>
```

Comment:
'|',
                     '&amp;',
                         ('project_id.privacy_visibility', '=', 'portal'),
                         ('project_id.message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id]),
                     '&amp;',
                         ('project_id.privacy_visibility', '=', 'portal'),
                          ('message_partner_ids', 'in', [user.partner_id.id])